community/logging/src/main/java/org/neo4j/logging/FormattedLog.java

```diff
@@ -203,19 +207,30 @@ protected void writeLog( PrintWriter out, String message )
         @Override
         protected void writeLog( PrintWriter out, String message, Throwable throwable )
         {
+            requireNonNull( message, "message" );
+
             lineStart( out );
             out.write( message );
-            out.write( ' ' );
-            out.write( throwable.getMessage() );
-            out.println();
-            throwable.printStackTrace( out );
+            if ( throwable != null )
+            {
+                if ( throwable.getMessage() != null )
```

Comment:
I agree class name should be logged and handling more things in `AbstractPrintWriterLogger` sounds like a good idea